src/main/java/com/conveyal/datatools/manager/jobs/MergeFeedsJob.java

```diff
@@ -0,0 +1,584 @@
+                                    int endDateIndex = getFieldIndex(fieldsFoundInZip, "end_date");
+                                    if (index == endDateIndex) {
+                                        LocalDate endDate = LocalDate.parse(csvReader.get(endDateIndex), GTFS_DATE_FORMATTER);
+                                        if (endDate.isAfter(LocalDate.now())) {
+                                            val = feedsToMerge.get(0).version.validationResult.firstCalendarDate
+                                                .minus(1, ChronoUnit.DAYS)
+                                                .format(GTFS_DATE_FORMATTER);
+                                        }
+                                    }
+                                }
+                                break;
+                            case "trips":
+                                // trip_ids between active and future datasets must not match. If any trip_id is found
+                                // to be matching, the merge should fail with appropriate notification to user with the
+                                // cause of the failure. Merge result should include all conflicting trip_ids.
+                                for (NewGTFSError error : idErrors) {
+                                    if (error.errorType.equals(NewGTFSErrorType.DUPLICATE_ID)) {
+                                        mergeFeedsResult.failureReasons.add("Trip ID conflict caused merge failure.");
+                                        mergeFeedsResult.idConflicts.add(error.badValue);
+                                        mergeFeedsResult.errorCount++;
+                                        if (failOnDuplicateTripId) mergeFeedsResult.failed = true;
+                                        skipRecord = true;
+                                    }
+                                }
+                                break;
+                            case "stops":
+                                // When stop_code is included, stop merging will be based on that. If stop_code is not
+                                // included, it will be based on stop_id. All stops in future data will be carried
+                                // forward and any stops found in active data that are not in the future data shall be
+                                // appended. If one of the feed is missing stop_code, merge fails with a notification to
+                                // the user with suggestion that the feed with missing stop_code must be fixed with
+                                // stop_code.
+                            case "routes":
+                                String primaryKeyValue = csvReader.get(table.getKeyFieldIndex(fieldsFoundInZip));
+                                Set<NewGTFSError> primaryKeyErrors = table.checkReferencesAndUniqueness(primaryKeyValue, lineNumber, field, val, referenceTracker);
```

Comment:
line is too long

Reply:
Since this is only used in certain circumstances, it should be moved within the conditional where it applies at line 447.

Reply:
Hrm, actually. I think I intentionally placed this here because when checking for references, it actually modifies sets contained within the reference tracker (i.e., it tracks them). If it only happens conditionally, then it would not track every ID and would not necessarily be able to discern whether a record is unique.